packages/twentytwenty-theme/src/components/archive/archive-pagination.js

```diff
@@ -2,72 +2,174 @@ import React, { useEffect } from "react";
   }, []);
 
   return (
-    <div>
-      {/* If there's a next page, render this link */}
-      {isThereNextPage && (
-        <Link link={nextPageLink}>
-          <Text>← Older posts</Text>
-        </Link>
-      )}
-
-      {isTherePreviousPage && isThereNextPage && " - "}
-
-      {/* If there's a previous page, render this link */}
-      {isTherePreviousPage && (
-        <Link link={prevPageLink}>
-          <Text>Newer posts →</Text>
-        </Link>
-      )}
-    </div>
+    <Container>
+      <div className="directions">
+        {hasNewerPosts && (
+          <Link link={getPageLink(page - 1)}>
+            ← <span className="dir">Newer</span>
+          </Link>
+        )}
+      </div>
+
+      <div id="generated">
+        <ul>
+          {paginationArray.map((item, index) => {
+            // if item is dots, "..."
+            if (item === "...") {
+              return (
+                <li className="dots" key={index}>
+                  ...
+                </li>
+              );
+            }
+
+            // if item is current page
+            if (item === page) {
+              return (
+                <li className="active" key={index}>
+                  {item}
+                </li>
+              );
+            }
+
+            return (
+              <li key={index}>
+                <Link link={getPageLink(item)}>{item}</Link>
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+
+      <div className="directions">
+        {hasOlderPosts && (
+          <Link link={getPageLink(page + 1)}>
+            <span className="dir">Older</span> →
+          </Link>
+        )}
+      </div>
+    </Container>
   );
 };
 
-/**
- * Connect Pagination to global context to give it access to
- * `state`, `actions`, `libraries` via props
- */
-export default connect(Pagination);
+const getMaxWidth = props => maxWidths[props.size] || maxWidths["medium"];
+
+const maxWidths = {
+  thin: "58rem",
+  small: "80rem",
+  medium: "100rem"
+};
+
+const Container = styled.div`
+  font-size: 1em;
+  font-weight: 600;
+  margin-left: auto;
+  margin-right: auto;
+  line-height: 30px;
+  width: calc(100% - 4rem);
+  max-width: ${getMaxWidth};
+
+  & > div {
+    display: inline-block;
+  }
+
+  @media (min-width: 700px) {
+    width: calc(100% - 8rem);
+    font-size: 1.3em;
+    font-weight: 700;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+
+    .directions {
+      a {
+        .dir::after {
+          content: " Posts";
+        }
+      }
+    }
+  }
 
-const Text = styled.em`
-  display: inline-block;
-  margin-top: 16px;
+  a {
+    text-decoration: none;
+
+    &:hover {
+      text-decoration: underline;
+    }
+  }
+
+  ul {
+    list-style: none;
+    margin: 0 2rem;
+
+    li {
+      display: inline-block;
+      margin: 0;
+
+      &:not(:last-of-type) {
+        margin-right: 2rem;
+      }
+    }
+  }
```

Comment:
Could you please migrate this CSS to Emotion by using either the `css` prop or styled components?

Reply:
Just to expand on what Luis meant: You should remove the nested CSS classes like `dir` or `directions` and create a styled component / use the[ `css`](https://emotion.sh/docs/css-prop) prop for each element that uses those CSS classes.